extensions/amp-story/0.1/utils.js

```diff
@@ -106,3 +106,64 @@ export function createShadowRootWithStyle(container, element, css) {
 
   return shadowRoot;
 }
+
+
+/**
+ * Parses the resolved CSS color property, that is always in the form of
+ * `rgba(0, 0, 0, 1)` or `rgb(0, 0, 0)`, that can be retrieved using
+ * `getComputedStyle`.
+ * Returns an object containing the R, G, and B 8bit numbers.
+ * @param  {string} cssValue
+ * @return {!Object<string, number>}
+ */
+export function getRGBFromCssColorValue(cssValue) {
+  const regexPattern = /rgba?\((\d{1,3}), (\d{1,3}), (\d{1,3})/;
+
+  if (!cssValue.match(regexPattern)) {
+    user().error('UTILS', 'getRGBFromCssColorValue expects a parameter in ' +
+        `the form of 'rgba(0, 0, 0, 1)' or 'rgb(0, 0, 0)' but got ${cssValue}`);
+    // Returns a fallback value, to fail 'gracefully' in case a browser we don't
+    // know about gave an unexpected value.
+    return {r: 0, g: 0, b: 0};
+  }
+
+  const matches = regexPattern.exec(cssValue);
+
+  return {
+    r: +matches[1],
+    g: +matches[2],
+    b: +matches[3],
+  };
+}
+
+
+/**
+ * Returns the color, either black or white, that has the best contrast ratio
+ * against the provided RGB 8bit values.
+ * @param  {!Object<string, number>} rgb  ie: {r: 0, g: 0, b: 0}
+ * @return {string} '#fff' or '#000'
+ */
+export function getTextColorForRGB({r, g, b}) {
```

Comment:
Optional: More of a UX question, but sometimes very high contrast between text and background color can be uncomfortable to read (e.g. `#fff` text on `#000` background). Since we're already doing the luminance calc anyways, we can instead try to use a gray color that satisfies a target contrast ratio. 

Material recommends 4.5 for small text: https://material.googleplex.com/usability/accessibility.html#accessibility-color-contrast

Reply:
Very good point!
I've been playing with this very cool tool: https://www.desmos.com/calculator/rbo4wzxoit
Pure white (``L = 1``) and pure black (``L = 0``) are actually the only two colors that always satisfy the minimum ratio of 4.5 required by the material guidelines (lowest is 4.6).
But you're right, we could do better at handling light text on dark background, which is easy to detect. We could try, for these high contrast ratio against dark backgrounds, using some gray color. I added a todo to handle this later